Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py

```diff
@@ -254,24 +247,87 @@
 }
 
 ERROR_DICT = {
-    '404': 'Invalid URL.',
-    '408': 'Invalid URL.',
-    '409': 'Invalid message or missing parameters.',
-    '500': 'Internal error.',
-    '503': 'Rate limit exceeded.'
+    404: 'Invalid URL.',
+    408: 'Invalid URL.',
+    409: 'Invalid message or missing parameters.',
+    500: 'Internal error.',
+    503: 'Rate limit exceeded.'
 }
 
 if PARAMS.get('mark_as_malicious'):
     verdicts = argToList(PARAMS.get('mark_as_malicious'))
-    for verdict in verdicts:
-        VERDICTS_TO_DBOTSCORE[verdict] = 3
+    VERDICTS_TO_DBOTSCORE.update(dict.fromkeys(verdicts, 3))
+
+DEFAULT_BUCKET_INFO = dict.fromkeys((
+    'minute_points',
+    'daily_points',
+    'minute_points_remaining',
+    'daily_points_remaining',
+    'minute_bucket_start',
+    'daily_bucket_start',
+), 'Unavailable')
+
+EXECUTION_METRICS = ExecutionMetrics()
+API_POINTS_TABLE = CommandResults(
+    outputs_prefix='AutoFocus.Quota',
+    replace_existing=True
+)
+
 
 ''' HELPER FUNCTIONS '''
 
 
+class RateLimitExceededError(BaseException):
+
+    def __init__(self, api_res: dict) -> None:
+        super().__init__()
+        self.api_res = api_res
+
+
+def return_metrics():
+    if EXECUTION_METRICS.metrics is not None and ExecutionMetrics.is_supported():
+        return_results(EXECUTION_METRICS.metrics)
+    if API_POINTS_TABLE.readable_output:
+        return_results(API_POINTS_TABLE)
+
+
+def rerun_command_if_required(api_res: dict, rate_limit_auto_retry: bool):
+    daily_points_remaining = dict_safe_get(api_res, ('bucket_info', 'daily_points_remaining'), 0)
+    if not (rate_limit_auto_retry and daily_points_remaining):
+        results = CommandResults(
+            readable_output=(
+                f'Error in API call to AutoFocus:\nMessage: {api_res.get("message")}\n'),
+            entry_type=EntryType.ERROR
+        )
+    else:
+        next_run_in_seconds = int(dict_safe_get(api_res, ('bucket_info', 'wait_in_seconds'), 70)) + 1  # type: ignore
```

Comment:
what if this number here is very high? waiting until tomorrow? We should check it cant be larger than a certain number (to be set as an advanced integration parameter with a default of five minutes)